Lib9c/Model/State/StakeState.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Nekoyume.Action;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Model.State
+{
+    public class StakeState : State
+    {
+        public class StakeAchievements
+        {
+            private readonly Dictionary<int, int> _achievements;
+
+            public StakeAchievements(Dictionary<int, int> achievements = null)
+            {
+                _achievements = achievements ?? new Dictionary<int, int>();
+            }
+
+            public StakeAchievements(Dictionary serialized)
+            {
+                _achievements = serialized.ToDictionary(
+                    pair => int.Parse(((Text) pair.Key).Value, CultureInfo.InvariantCulture),
+                    pair => (int) (Integer) pair.Value);
+            }
+
+            public IValue Serialize() =>
+                new Dictionary(_achievements.OrderBy(pair => pair.Key).Select(pair =>
```

Comment:
IMO, it would be better to serialize it as a List rather than a Dictionary.

Reply:
Currently, It seems like `Dictionary[Level => Step]` so you think like `List[Tuple[Level => Step]]`?
As a comment, the reason to call `OrderBy` explicitly there though Bencodex `Dictionary` guarantees its order, it is because Libplanet Analyzer warned it though it isn't misusage.

<img width="1199" alt="image" src="https://user-images.githubusercontent.com/26626194/165260062-fd0638df-ed8c-4232-8f63-00f0e1d16405.png">

Reply:
I rewrite them like the below lines. If you think like that because it has order, how about them?

```csharp
            public IValue Serialize() =>
                new Dictionary(_achievements.ToDictionary(
                    pair => (IKey)(Text) pair.Key.ToString(CultureInfo.InvariantCulture),
                    pair => (IValue)(Integer) pair.Value));
```

Reply:
At this time, I'll merge this pull request 😞 Thanks for your opinions and reviews!